Clarified RBAC doc about resourceNames field and create verb (#50455)
kubernetes/website: content/en/docs/reference/access-authn-authz/rbac.md

```

{{< note >}}
You cannot restrict **deletecollection** or top-level **create** requests by resource name.

> For **create**, this limitation is because the name of the new object may not be known at authorization time. However, the **create** limitation applies only to top-level resources, not subresources. For example, you can use the `resourceNames` field with `pods/exec`.

Reviewer: We believe you can actually restrict this if you use Server side apply (SSA) for create. Could you try investigating that and updating the docs to cover that as well?

Reviewer 2: (it's also OK to file a new issue about SSA / `resourceNames`, and park this PR whilst that issue and detail gets addressed)

Author: I tried to apply my tests resources declared in the issue i openned. I doesn't works either

If you restrict **list** or **watch** by `resourceName`, clients must include a `metadata.name` field selector in their **list** or **watch** request (that matches the specified `resourceName`)
in order to be authorized.

> For example: `kubectl get configmaps --field-selector=metadata.name=my-configmap`

Reviewer 2: ```suggestion
For example: `kubectl get configmaps --field-selector=metadata.name=my-configmap --watch`
```
?

Restricting by resource name feels more relevant to a watch. Not sure if that's true but the plain **list** case does feel contrived.

{{< note >}}

> You cannot restrict `create` or `deletecollection` requests by their resource name.

Reviewer 2: ```suggestion
You cannot restrict **deletecollection** or object-level **create** requests by resource name.
```

> For `create`, this limitation is because the name of the new object may not be known at authorization time. However, the `create` limitation applies only to top-level resources, not subresources. For example, you can use the `resourceNames` field with `pods/exec`.
> If you restrict `list` or `watch` by resourceName, clients must include a `metadata.name` field selector in their `list` or `watch` request that matches the specified resourceName in order to be authorized.
> For example, `kubectl get configmaps --field-selector=metadata.name=my-configmap`

Reviewer 2: We write API verbs in lowercase boldface, so:
```suggestion
For **create**, this limitation is because the name of the new object may not be known at authorization time. However, the **create** limitation applies only to top-level resources, not subresources. For example, you can use the `resourceNames` field with `pods/exec`.
If you restrict **list** or **watch** by `resourceName`, clients must include a `metadata.name` field selector in their **list** or **watch** request (that matches the specified `resourceName`)
in order to be authorized.
For example: `kubectl get configmaps --field-selector=metadata.name=my-configmap`
```
